src/AdminConsole/Components/Pages/Organization/SettingsComponents/DeleteOrganizationComponent.razor.cs

```diff
@@ -0,0 +1,91 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Forms;
+
+namespace Passwordless.AdminConsole.Components.Pages.Organization.SettingsComponents;
+
+public partial class DeleteOrganizationComponent : ComponentBase
+{
+    public const string DeleteFormName = "delete-form";
+
+    /// <summary>
+    /// Organization Name
+    /// </summary>
+    [Parameter]
+    public required string Name { get; set; }
+
+    /// <summary>
+    /// The amount of active applications belonging to the organization.
+    /// </summary>
+    [Parameter]
+    public required int ApplicationsCount { get; set; }
+
+    [SupplyParameterFromForm(FormName = DeleteFormName)]
+    public DeleteFormModel DeleteForm { get; set; } = new();
+
+    public EditContext? DeleteFormEditContext { get; set; }
+
+    public ValidationMessageStore? DeleteFormValidationMessageStore { get; set; }
+
+    /// <summary>
+    /// Whether the organization can be deleted.
+    /// </summary>
+    public bool CanDelete => ApplicationsCount == 0;
+
+    protected override void OnInitialized()
+    {
+        DeleteFormEditContext = new EditContext(DeleteForm);
+        DeleteFormValidationMessageStore = new ValidationMessageStore(DeleteFormEditContext);
+    }
+
+    private async Task OnDeleteFormSubmittedAsync()
+    {
+        if (!CanDelete)
```

Comment:
We did not have this check here, could have been a backdoor to delete an organization by posting the form directly.